src/main/java/fiji/plugin/trackmate/gui/TrackMateGUIController.java

```diff
@@ -219,6 +223,21 @@ public TrackMateGUIController( final TrackMate trackmate )
 		this.gui = new TrackMateWizard( this );
 		this.logger = gui.getLogger();
 
+		/*
+		 * Add this TrackMate instance to the ObjectService
+		 */
+		ObjectService objectService = TMUtils.getContext().service( ObjectService.class );
+		if ( objectService != null )
+		{
+			ImagePlus imp = trackmate.getSettings().imp;
+			String name = gui.getTitle();
+			if ( imp != null )
+			{
+				name += " (" + imp.getTitle() + ")";
+			}
+			objectService.addObject( trackmate, name );
+		}
+
```

Comment:
As you suggested let's go for the `Named` interface.
Also the name should incorporate the `hashCode` in case we have 2 TrackMates running on the same image.

Reply:
👍